source/contracts/extensions/MarketFeeCalculator.sol

```diff
@@ -3,45 +3,76 @@ pragma solidity ^0.4.13;
-        IReportingWindow _previousReportingWindow = _universe.getReportingWindowByTimestamp(_previousTimestamp);
+        IReportingWindow _previousReportingWindow = _reportingWindow.getPreviousReportingWindow();
+        uint256 _totalMarketsInPreviousWindow = _reportingWindow.getNumMarkets();
+        uint256 _indeterminateMarketsInPreviousWindow = _reportingWindow.getNumIndeterminateMarkets();
         uint256 _previousValidityBondInAttoeth = validityBondInAttoeth[_previousReportingWindow];
-        if (_previousValidityBondInAttoeth == 0) {
-            _previousValidityBondInAttoeth = 1 * 10 ** 16;
-        }
-        uint256 _targetIndeterminateMarketsPerHundred = 1;
-        _currentValidityBondInAttoeth = _previousValidityBondInAttoeth * _targetIndeterminateMarketsPerHundred * _totalMarketsInPreviousWindow / _indeterminateMarketsInPreviousWindow / 100;
+
+        _currentValidityBondInAttoeth = calculateValidityBond(_indeterminateMarketsInPreviousWindow, _totalMarketsInPreviousWindow, TARGET_INDETERMINATE_MARKETS_DIVISOR, _previousValidityBondInAttoeth);
         validityBondInAttoeth[_reportingWindow] = _currentValidityBondInAttoeth;
         return _currentValidityBondInAttoeth;
     }
 
-    function getTargetReporterGasCosts() constant public returns (uint256) {
-        // TODO: get number of registration tokens issued last period
-        // TODO: get target reporter count + wiggle room
-        // TODO: calculate estimated reporters per market
-        uint256 _estimatedReportsPerMarket = 10;
-        // TODO: figure out what the number actually is
-        uint256 _gasToReport = 100000;
-        // we double it to ensure we have more than enough rather than not enough
-        uint256 _estimatedReportingGas = _gasToReport * _estimatedReportsPerMarket * 2;
-        // TODO: multiply this by average gas costs of reporters historically
-        return _estimatedReportingGas;
+    function calculateValidityBond(uint256 _indeterminateMarkets, uint256 _totalMarkets, uint256 _targetIndeterminateMarketsDivisor, uint256 _previousValidityBondInAttoeth) constant public returns (uint256) {
+        if (_totalMarkets == 0) {
+            return DEFAULT_VALIDITY_BOND;
+        }
+        if (_previousValidityBondInAttoeth == 0) {
+            _previousValidityBondInAttoeth = DEFAULT_VALIDITY_BOND;
+        }
+        
+        uint256 _targetIndeterminateMarkets = _totalMarkets.div(_targetIndeterminateMarketsDivisor);
+
+        if (_indeterminateMarkets <= _targetIndeterminateMarkets) {
+            return _indeterminateMarkets
+                .mul(_previousValidityBondInAttoeth)
+                .mul(_targetIndeterminateMarketsDivisor)
+                .div(_totalMarkets)
+                .div(2)
+            .add(_previousValidityBondInAttoeth.div(2));
+        } else {
+            return _targetIndeterminateMarketsDivisor
+                .mul(_previousValidityBondInAttoeth.mul(_indeterminateMarkets)
```

Comment:
Recommend each operation on its own line, or all operations on the same line.  In this case, I think all on own line is more readable.

Reply:
solium doesn't seem to like that as the previous line doesn't end in a paren when this is done. Any solium tricks for this one?

Reply:
Oh, bleh.  Didn't notice the open without close.  Does it complain if we indent the following lines?  I'm now more concerned about the fact that it was so easy to miss the open here followed by a close several lines down.  Ideally I would prefer something like:
```
_targetIndeterminateMarketsDivisor
    .mul(_previousValidityBondInAttoeth
        .mul(_indeterminateMarkets)
        .div(_totalMarkets)
        .sub(_previousValidityBondInAttoeth
            .div(_targetIndeterminateMarketsDivisor)
        )
    )
    .div(_targetIndeterminateDivisor - 1)
    .add(_previousValidityBondInAttoeth);
```
Though, after typing that all out I have to wonder if there is some value in some intermediate variables for the nested section?  Problem of course is that we don't have a good name for them as they are just transient pieces of the formula... The advantage of the above, while awkward to type and read, is that it makes it clear when the nested math is happening so bad readers like me don't fail to miss the missing close parens.

Reply:
Yea solium dislikes extra indentation as well. As a bonus this function is close to stack depth problems so transient variables probably cant be an option either